metashare/api/views.py

```diff
@@ -80,8 +81,15 @@ class RepositoryViewSet(viewsets.ModelViewSet):
     model = Repository
 
     def get_queryset(self):
-        repositories = self.request.user.repositories.values_list("url", flat=True)
-        return Repository.objects.filter(repo_url__in=repositories)
+        repo_ids = self.request.user.repositories.values_list("repo_id", flat=True)
+
+        for repo in Repository.objects.filter(repo_id__isnull=True):
+            try:
+                repo.get_repo_id(self.request.user)
+            except ResponseError:
+                pass
```

Comment:
So I'd guess this is probably frowned upon? It forces an external API call in-band for any repos which don't have IDs yet -- on every `get_queryset`... 😬

Reply:
Yeah, that's a little upsetting. Let's… see if it is a problem maybe?

Reply:
This _should_ only happen once ever per repo in the DB. So I would think in practice it won't be an issue? But let's discuss.